contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js

```diff
@@ -60,9 +43,11 @@ export function loadChannelTree(context, channel_id) {
 }
 
 export function loadTrashTree(context, channelId) {
-  return getChannel(context, channelId).then(channel =>
-    context.dispatch('loadTree', { tree_id: channel.trash_root_id })
-  );
+  const channel = context.rootGetters['channel/getChannel'](channelId);
```

Comment:
In general I would suggest not using much dependencies on getters inside of actions whenever possible and rather provided all required information from the place that calls an action through parameters to make things more explicit otherwise it will become difficult to track all dependencies to data that is required for an action (here it would mean rather passing `tree_id` as a parameter instead of `channelId`). Another solution would be to prefetch all required data but that might result in too many unnecessary API calls. I didn't want to spend too much time updating views because I might need to update these places soon and also don't consider it be a crucial issue - so for now I've just added an `Error` to be thrown and wanted to mention here that we will need to be careful about similar use cases in general.

Reply:
Yeah I think moving towards accepting a `tree_id` makes sense. I do think this change requires an update to the [TrashModal](https://github.com/learningequality/studio/blob/vue-refactor/contentcuration/contentcuration/frontend/channelEdit/views/trash/TrashModal.vue#L216).

Reply:
Okay, makes sense to me - I saw your suggestions on Slack in studio-refactor so will try to go in this direction and will check it in scope of staging channels work as its related.